src/components/Divider.tsx

```diff
@@ -1,70 +1,182 @@
+  return thoughts.map(thought => {
+    const innerThoughtElement = document.querySelector(
+      `[aria-label="tree-node"][data-id="${thought.id}"] [aria-label="thought"]`,
+    )
+
+    // Use DOM widths for thoughts
+    return innerThoughtElement?.getBoundingClientRect().width ?? 0
+  })
+}
 
 /** A custom horizontal rule. */
 const Divider = ({ path, cssRaw }: { path: Path; cssRaw?: SystemStyleObject }) => {
-  const dividerSetWidth = React.createRef<HTMLInputElement>()
   const dispatch = useDispatch()
+  const dividerRef = useRef<HTMLDivElement>(null)
+  const [dividerWidth, setDividerWidth] = useState<number>(DIVIDER_MIN_WIDTH)
+  const { dividerId, isOnlyChild, isTableView, children, thoughtsAtSameDepth } = useDividerData(path)
+  const resizeObservers = useRef<ResizeObserver[]>([])
 
   /** Sets the cursor to the divider. */
   const setCursorToDivider = (e: React.MouseEvent | React.TouchEvent) => {
     e.stopPropagation()
     dispatch(setCursor({ path }))
   }
 
-  /** Get the max width of nearby for divider list child elements, add 30 px and set this width for divider. */
-  const setStyle = () => {
-    if (dividerSetWidth.current) {
-      const parentUl = dividerSetWidth.current.closest('ul')
-      const children = parentUl ? (Array.from(parentUl.childNodes) as HTMLElement[]) : []
-      const widths = children.map((child: HTMLElement) => {
-        if (child.querySelector('[data-divider=true]')) return DIVIDER_PLUS_PX
-        const subs = child.getElementsByClassName('subthought') as HTMLCollectionOf<HTMLElement>
-        return subs.length ? subs[0].offsetWidth + DIVIDER_PLUS_PX : DIVIDER_PLUS_PX
+  useEffect(() => {
+    /** Calculates and updates the Divider's width based on sibling thought widths. */
+    const updateDividerWidth = () => {
+      if (!dividerRef.current) return
+
+      if (isOnlyChild && !isTableView) {
+        setDividerWidth(DIVIDER_MIN_WIDTH)
+        return
+      }
+
+      let widths: number[] = []
+      if (isTableView && isOnlyChild) {
+        if (thoughtsAtSameDepth.length === 0) {
+          setDividerWidth(DIVIDER_MIN_WIDTH)
+          return
+        }
+        widths = getThoughtWidths(thoughtsAtSameDepth)
+      } else {
+        // Non-Table View or Divider is not the only item
+        const siblingThoughts = children.filter(child => child.id !== dividerId && !isDivider(child.value))
+
+        if (siblingThoughts.length === 0) {
+          setDividerWidth(DIVIDER_MIN_WIDTH)
+          return
+        }
+        widths = getThoughtWidths(siblingThoughts)
+      }
+
+      setDividerWidth(Math.round(Math.max(...widths, DIVIDER_MIN_WIDTH)))
+    }
+
+    updateDividerWidth() // Initial call to set the width
+
+    /** Attach resize observers to relevant thought elements. */
+    const observeResize = () => {
+      // Clear any existing observers
+      resizeObservers.current.forEach(observer => observer.disconnect())
+      resizeObservers.current = []
```

Comment:
Do we need to disconnect each observer and empty the array here? This seems redundant since it's already handled in the returned cleanup function.